src/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs

```diff
@@ -0,0 +1,167 @@
+
+            DependencyContext context = null;
+
+            if (IsEntryAssembly(assembly))
+            {
+                context = LoadEntryAssemblyContext();
+            }
+
+            if (context == null)
+            {
+                context = LoadAssemblyContext(assembly);
+            }
+
+            if (context?.IsPortable == true)
+            {
+                var runtimeContext = LoadRuntimeContext();
+                if (runtimeContext != null)
+                {
+                    context = context.Merge(runtimeContext);
+                }
+            }
+            return context;
+        }
+
+        private DependencyContext LoadEntryAssemblyContext()
+        {
+            if (!string.IsNullOrEmpty(_entryPointDepsLocation))
+            {
+                Debug.Assert(File.Exists(_entryPointDepsLocation));
+                using (var stream = _fileSystem.File.OpenRead(_entryPointDepsLocation))
+                {
+                    return _jsonReader.Read(stream);
+                }
+            }
+            return null;
+        }
+
+        private DependencyContext LoadRuntimeContext()
+        {
+            if (!string.IsNullOrEmpty(_runtimeDepsLocation))
+            {
+                Debug.Assert(File.Exists(_runtimeDepsLocation));
+                using (var stream = _fileSystem.File.OpenRead(_runtimeDepsLocation))
+                {
+                    return _jsonReader.Read(stream);
+                }
+            }
+            return null;
+        }
+
+        private DependencyContext LoadAssemblyContext(Assembly assembly)
+        {
+            using (var stream = GetResourceStream(assembly, assembly.GetName().Name + DepsJsonExtension))
+            {
+                if (stream != null)
+                {
+                    return _jsonReader.Read(stream);
+                }
+            }
+
+            var depsJsonFile = Path.ChangeExtension(assembly.Location, DepsJsonExtension);
+            if (_fileSystem.File.Exists(depsJsonFile))
+            {
+                using (var stream = _fileSystem.File.OpenRead(depsJsonFile))
+                {
+                    return _jsonReader.Read(stream);
+                }
+            }
+
+            var depsFile = Path.ChangeExtension(assembly.Location, DepsExtension);
+            if (_fileSystem.File.Exists(depsFile))
+            {
+                using (var stream = _fileSystem.File.OpenRead(depsFile))
+                {
+                    return _csvReader.Read(stream);
+                }
+            }
+
+            return null;
+        }
+
+        private static string GetDefaultRuntimeDepsLocation()
+        {
+            var runtimeLocation = Path.GetDirectoryName(typeof(object).Assembly.Location);
+            if (runtimeLocation != null)
+            {
+                // HACK: searches for single deps.json file next to mscorlib
+                var depsFiles = Directory.GetFiles(runtimeLocation, "*" + DepsJsonExtension);
```

Comment:
/cc @schellap do we know what the name of the runtime's deps file will be? I assume `corehost` requires it, so you might know the name :)